tools/lint/clang_format.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+import argparse
+import sys
+import os
+import re
+import logging
+from utils import (
+    run_command,
+    init_environment,
+    append_file,
+    emit_problem_matchers,
+    write_file,
+    add_common_arguments,
+)
+
+
+def count_clang_format_errors(log_text: str) -> int:
+    """
+    Count lines that indicate clang-format violations.
+    The pattern looks for lines ending with "[-Wclang-format-violations]".
+    """
+    pattern = r"\[-Wclang-format-violations\]$"
+    matches = re.findall(pattern, log_text, re.MULTILINE)
+    logging.debug(f"Matches found: %s", matches)
```

Comment:
If we use f-string, let's use f-string
```suggestion
    logging.debug(f"Matches found: {matches}")
```